printer.go

```diff
@@ -126,6 +126,26 @@ func Default() (string, error) {
 	return syscall.UTF16ToString(b), nil
 }
 
+const maxStringSize = 1 << 20
+
+func utf16PtrToString(p *uint16) string {
+	if p == nil {
+		return ""
+	}
+	// Find NUL terminator.
+	end := unsafe.Pointer(p)
+	n := 0
+	for *(*uint16)(end) != 0 {
+		end = unsafe.Pointer(uintptr(end) + unsafe.Sizeof(*p))
+		n++
+	}
+	if n >= maxStringSize {
+		return ""
```

Comment:
Perhaps this should be an error. Does make the function a pain to use as can't use it in a single line assignment. Unclear what the error handling should be either - just skip the field or fail the whole call?